src/qibo_client/qibo_job.py

```diff
@@ -0,0 +1,221 @@
+import tarfile
+import tempfile
+import time
+import typing as T
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+import qibo
+import requests
+
+from . import constants
+from .config_logging import logger
+from .utils import QiboApiRequest
+
+
+def convert_str_to_job_status(status: str):
+    return next((s for s in QiboJobStatus if s.value == status), None)
+
+
+class QiboJobStatus(Enum):
+    QUEUED = "to_do"
+    RUNNING = "in_progress"
+    DONE = "success"
+    ERROR = "error"
+
+
+def wait_for_response_to_get_request(
```

Comment:
Do you really need this function outside the `QiboJob` class? If you move there as a private function `_wait_for_response...` you can access the status and avoid all the problems raised in the discussion above.